pygmt/tests/test_clib_virtualfiles.py

```diff
@@ -0,0 +1,404 @@
+"""
+Test the API functions related to virtual files.
+"""
+import os
+from contextlib import contextmanager
+from itertools import product
+
+import numpy as np
+import pandas as pd
+import pytest
+import xarray as xr
+from pygmt import clib
+from pygmt.exceptions import GMTCLibError, GMTInvalidInput
+from pygmt.helpers import GMTTempFile
+
+TEST_DATA_DIR = os.path.join(os.path.dirname(__file__), "data")
+POINTS_DATA = os.path.join(TEST_DATA_DIR, "points.txt")
+
+
+@pytest.fixture(scope="module", name="dtypes")
+def fixture_dtypes():
+    """
+    List of supported numpy dtypes.
+    """
+    return "int8 int16 int32 int64 uint8 uint16 uint32 uint64 float32 float64".split()
+
+
+@pytest.fixture(scope="module", name="data")
+def fixture_data():
+    """
+    Load the point data from the test file.
+    """
+    return np.loadtxt(POINTS_DATA)
```

Comment:
Small nit, the data fixture was originally placed before the dtypes fixture :upside_down_face::

```suggestion
@pytest.fixture(scope="module", name="data")
def fixture_data():
    """
    Load the point data from the test file.
    """
    return np.loadtxt(POINTS_DATA)


@pytest.fixture(scope="module", name="dtypes")
def fixture_dtypes():
    """
    List of supported numpy dtypes.
    """
    return "int8 int16 int32 int64 uint8 uint16 uint32 uint64 float32 float64".split()
```